macos/test_app.rb

```diff
@@ -0,0 +1,14 @@
+#
+# Copyright (c) Microsoft Corporation. All rights reserved.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+#
+
+require_relative('../ios/test_app.rb')
+
+def use_test_app!
+  use_test_app_internal!(:macos) do |target|
+    yield target if block_given?
+  end
+end
```

Comment:
FYI You can also capture and pass on a block like so:

```ruby
def use_test_app!(&block)
  use_test_app_internal!(:macOS, &block)
end
```

Reply:
Wouldn't this also make `block` a required parameter? I tried this originally but couldn't get it to work properly with `yield` inside of `use_test_app_internal!`.

Reply:
It shouldn’t, although my experience might be outdated. Will verify when I get back to my machine, but please feel free to merge in the meantime 👍

Reply:
Seems to work, on 2.6 at least:

```
irb(main):001:0> def internal
irb(main):002:1>   yield :ok if block_given?
irb(main):003:1> end
=> :internal
irb(main):004:0> def external(&block)
irb(main):005:1>   internal(&block)
irb(main):006:1> end
=> :external
irb(main):007:0> external
=> nil
irb(main):008:0> external { |x| p x }
:ok
=> :ok
```

Reply:
This is what I'm seeing :/

```
% ruby --version
ruby 2.7.1p83 (2020-03-31 revision a0c7c23c9c) [x86_64-darwin19]
% pod install

[!] Invalid `Podfile` file: no block given (yield).

 #  from /~/react-native-test-app/example/ios/Podfile:5
 #  -------------------------------------------
 #
 >  use_test_app!
 #  -------------------------------------------
```

The diff:

```diff
diff --git a/example/ios/Podfile b/example/ios/Podfile
index 59f8153..503f32a 100644
--- a/example/ios/Podfile
+++ b/example/ios/Podfile
@@ -2,8 +2,4 @@ require_relative '../node_modules/react-native-test-app/test_app.rb'

 workspace 'Example.xcworkspace'

-use_test_app! do |target|
-  target.tests do
-    pod 'Example-Tests', :path => '..'
-  end
-end
+use_test_app!
diff --git a/test_app.rb b/test_app.rb
index a1b3ba6..e45d559 100644
--- a/test_app.rb
+++ b/test_app.rb
@@ -7,8 +7,6 @@

 require_relative('ios/test_app.rb')

-def use_test_app!
-  use_test_app_internal!(:ios) do |target|
-    yield target if block_given?
-  end
+def use_test_app!(&block)
+  use_test_app_internal!(:ios, &block)
 end
```

Reply:
I think that’s because you also removed this, no?

```
use_test_app! do |target|
  target.tests do
    pod 'Example-Tests', :path => '..'
  end
end
```

Reply:
The block passed to `use_test_app!` should be optional, otherwise we'd end up with users having to do something like
```ruby
use_test_app! do |target|
end
```
Hence why I asked whether the block can still be optional. Sorry for not being clearer.

Reply:
Ah right, no that’s my bad; you did say so and that’s also why I tested it in my snippet. I think I understand the problem, you should move the `if block_given?` check into `use_test_app_internal` https://github.com/microsoft/react-native-test-app/blob/1a39ae21142d5b05b8f9b51e1fa6abdd9c30cc9b/ios/test_app.rb#L139

Reply:
Ahh, right… I forgot that I moved it out. Seems to work now. No idea what issue I hit earlier :/

Anyway: #93